docs/reference/api/connectivity/networksocket/networksocket.md

```diff
@@ -43,15 +43,69 @@ sock.recv(buf, 100);
 sock.close();
 ```
 
-### Network socket classes
+### Changes in Mbed OS 5.10
+
+Mbed OS Socket API has gone under refactoring for Mbed OS 5.10 release. For most of the applications, these changes are not noticeable as legacy behaviour is still emulated in `TCPSocket` and `UDPSocket` classes.
+
+New design contain Abstract Socket interface that can be used directly by
+applications. Casting `Socket` pointers back to `TCPSocket` or `UDPSocket`
+is no longer necessary.
+
+Downcasting any protocol specific class to `Socket` has no side effect and is recommended API design. Knowing the exact type is only required when socket is created.
```

Comment:
That's actually upcasting.

Reply:
Good catch. I always thought these terms as a opposite direction.
Down -> towards the base, as base is..bottom..
but actually in UML diagrams, it is up.

Reply:
I didn't actually know for sure - was doubtful and had to look it up before commenting.